app/models/user.rb

```diff
@@ -186,6 +186,7 @@ class User < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :created_podcasts, class_name: "Podcast", foreign_key: :creator_id, inverse_of: :creator, dependent: :nullify
   has_many :credits, dependent: :destroy
+  has_many :discussion_locks, dependent: :destroy, inverse_of: :locking_user, foreign_key: :locking_user_id
```

Comment:
Is it even possible to destroy the user if they still have discussion locks active due to the FK constraint? If ActiveRecord doesn't apply a `DEFERRED` clause to the FK constraint, this line may not have any effect. Should we set the FK to `ON DELETE CASCADE` instead?

That brings up a product-related question, actually. Does it make sense to silently unlock discussions simply because the user that locked them was deleted? Or would it make sense instead to have some kind of "ghost" user like GitHub does, where whenever a user is deleted, all their comments, PRs, etc, all get assigned to the ghost user?

Reply:
Right, but I mean, will Postgres even _allow_ us to delete that user with the FK constraint in place?

Reply:
Huh, TIL ActiveRecord destroys the associated records _before_ destroying the parent record. I thought the `dependent: :destroy` stuff was run with `after_destroy` hooks, but apparently it's run with the `before_destroy`. 💡 

```
  DiscussionLock Destroy (0.2ms)  DELETE FROM "discussion_locks" WHERE "discussion_locks"."id" = $1  [["id", 5]]
 <snip>
  User Destroy (7.9ms)  DELETE FROM "users" WHERE "users"."id" = $1  [["id", 9]]
```